### DISCUSSION TITLE
Image and File Fields with S3 Signed URL

### DISCUSSION THREAD

**Discussion by lilianlaupt** [laravel/nova-issues#6532] [Q&A]
- Laravel Version: v10.48.20
- Nova Version: 4.35.0
- PHP Version: 8.1
- Operating System and Version: Mac and Ubuntu
- Browser type and version: Chrome 

### Description:
I'm working on a project that utilizes Laravel as the backend, Nova as the web panel (CRUD), and a mobile app for end-users. The mobile app allows users to upload photos, while staff members can send information with attachments (images and PDFs) to these users.

Currently, all photos and attachments are stored on the application server. We would like to migrate this storage to S3 for improved scalability and security.

I understand that Nova supports S3 disk, but I want to keep my S3 bucket private. Therefore, I need to generate signed temporary URLs for Nova to display and download images and files. However, I haven't been able to find any documentation or resources on how to achieve this.

I've tried returning absolute URLs in my resource attributes, but the field always adds a '/storage' or AWS S3 host in front of the URL, preventing the images from displaying correctly.

I would appreciate some guidance on how to effectively use S3 signed URLs with Nova for displaying and downloading images and files.

**Comment by crynobone** [comment]
Nova uses VaporFile and VaporImage to handle S3 signed URL
